src/pages/posts/build-your-own-probot.md

```diff
@@ -0,0 +1,408 @@
+
+![Probot webhook handling flow](/images/probot-flow.png)
+
+First, it verifies the webhook signature; along with the JSON payload, GitHub sends an `X-GitHub-Secret` header. The value of the header is a combination of a secret key and the contents of the payload itself. GitHub and your Probot app both have the secret key (Probot uses the `WEBHOOK_SECRET` environment variable), so when the two services generate the header they should match exactly. If they don't, Probot ignores the request.
+
+This is a security measure to ensure that random POST requests aren't acted upon - only GitHub can trigger your app. This logic is now abstracted in a separate module that Probot uses, [`@octokit/webhooks`](https://github.com/octokit/webhooks.js), for convenience and reusability.
+
+Once the webhook verification is complete, Probot emits an event through its internal [`EventEmitter`](https://nodejs.org/api/events.html):
+
+```js
+const express = require('express')
+const EventEmitter = require('events')
+const app = express()
+
+const events = new EventEmitter()
+
+app.post('/', (req, res) => {
+  // Verify the webhook payload
+  const isValidWebhook = verifyWebhookPayload(req)
+  // The name of the GitHub event is sent in a header
+  const eventName = req.get('X-GitHub-Event')
+  // It's valid, we emit an event
+  if (isValidWebhook) {
+    events.emit('github-event', {
+      event: eventName,
+      payload: req.body
+    })
+    // We tell GitHub that everything worked great!
+    return res.send(200)
+  } else {
+    // Couldn't validate the webhook, so we send an "Unauthorized" code
+    return res.send(401)
+  }
+})
+```
+
+If your app is deployed to a public URL, GitHub can make an HTTP request; but if you're just working locally, you'll have to use something like `localtunnel` or `ngrok` to expose your local machine to the internet. We built [Smee.io](https://smee.io) specifically for Probot, to receive webhook events locally - it's its own topic, so if you're interested in a post on Smee [let me know](https://twitter.com/JasonEtco)!
+
+Somewhere behind the scenes, Probot has registered the event handlers of your Probot app. A standard app looks like this:
+
+```js
+module.exports = app => {
+  app.on('event', handler)
+}
+```
+
+Where `app` is an instance of the `Application` class, which is (sort of) an extension of `EventEmitter! We'll look at that shortly - there are still a things that Probot does before running your code, but first...
+
+### Why an EventEmitter API?
+
+To me, it's the inherent simplicity of "something happened, so now run this code." When an issue is opened on GitHub, run this code. This let's you focus on your app code, and not care what Probot is doing under the hood. Similar to something like Express's `app.get()`, but this let's you register multiple event handlers for the same event, and have an infinite list of events (since they're just strings).
```

Comment:
```suggestion
To me, it's the inherent simplicity of "something happened, so now run this code." When an issue is opened on GitHub, run this code. This lets you focus on your app code, and not care what Probot is doing under the hood. Similar to something like Express's `app.get()`, but this lets you register multiple event handlers for the same event, and have an infinite list of events (since they're just strings).
```